lib/Data/Chronicle/Reader.pm

```diff
@@ -93,6 +95,11 @@ has [qw(cache_reader db_handle)] => (
     default => undef,
 );
 
+#Alternatively you can pass a memory-based hash which will be used to fetch data
+has memory_map => (
```

Comment:
Since this is only fetching data, not writing data, why not pass it as the `cache_reader` parameter? Introducing a new field like this adds some confusion. Is it valid to have both `cache_reader` and `memory_map`? What format is the data in `memory_map`? What populates it?

I'm also concerned that there's no documentation - this is a public module, so if we're adding new features, they need to be documented and listed in the `Changes` file.